edfio/_header_field.py

```diff
@@ -30,20 +29,9 @@ def encode_int(value: int, length: int) -> bytes:
 
 
 def encode_float(value: float, length: int) -> bytes:
-    value_str = str(value)
-    if len(value_str) > length:
-        integer_part_length = len(str(int(value)))
-        if value < 0 and int(value) == 0:
-            integer_part_length = 2
-        if integer_part_length > length:
-            raise ValueError(f"{value} exceeds maximum field length {length}")
-        if integer_part_length == length:
-            value_str = f"{value:.0f}"
-        else:
-            value_str = f"{value:.{length-1-integer_part_length}f}"
-        msg = f"{value} exceeds maximum field length {length}, rounding to {value_str}"
-        warnings.warn(msg)
-    return encode_str(value_str, length)
+    if float(value).is_integer():
+        value = int(value)
+    return encode_str(str(value), length)
```

Comment:
Is it possible that str(value) generates a string longer than 8 characters? If yes, encode_str will fail later on.

Reply:
It is! But that is desired (e.g. to disallow setting a data_record_duration that would exceed the field size) and consistent with `encode_int`. When setting the physical min and max, `round_float_to_8_characters` ensures this won't fail.